api/kroxylicious-filter-api/pom.xml

```diff
@@ -52,6 +55,7 @@
         <dependency>
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-api</artifactId>
+            <scope>provided</scope>
```

Comment:
Little unsure here about how opinionated we should be, slf4j-api is quite a light dependency to propagate and I'd consider filter logging core functionality (micrometer metrics less so).

Reply:
I think I was concerned about forcing slf4j 2.x onto people. As its an API jar bundling it seems less of an issue.

Reply:
Ah, so you could imagine someone using slf4j 1.x, maybe with test dependencies set up the way they like with a 1.x backend, but it will work fine when dropped into kroxylicious since slf4j is backwards compatible.